### PR TITLE
Reload/Interaction - Fix 2 hemtt ingame errors

### PR DESCRIPTION
**Pull request by jokoho48**
**When merged this pull request will:**
- Fixes 2 issues that occur when building with hemtt

### CODE REVIEW

File: addons/reload/CfgEventHandlers.hpp [hunk 1]
```diff
@@ -27,8 +27,7 @@ class Extended_Take_EventHandlers {
                     && {GVAR(DisplayText)} \
                     && {_container in [ARR_3(uniformContainer _unit, vestContainer _unit, backpackContainer _unit)]} \
                     && {_item == currentMagazine _unit} \
-                ) then {_unit call DFUNC(displayAmmo)}; \
-            );
+                ) then {_unit call DFUNC(displayAmmo)};);
```
Comment (BaerMitUmlaut) [inline addons/reload/CfgEventHandlers.hpp:30]: Why is this a DFUNC?

**Review by bux — CHANGES_REQUESTED** [review]
Please split the PR in two and add proper titles for the changelog describing the issue in more detail

File: addons/reload/CfgEventHandlers.hpp (hunk 1 above)
Comment (jokoho48) [reply addons/reload/CfgEventHandlers.hpp:30]: dont ask me :P i just fixed the general error that happend when you build it with hemtt

**Comment by jokoho48** [comment]
> Please split the PR in two and add proper titles for the changelog describing the issue in more detail

tbh I think this should be excluded from the changelog because it is not a thing that the user really interests or is currently an issue for the builds we release.

**Comment by veteran29** [comment]
I don't like the idea of making the code less readable because the build tool fails, I guess it's hard to fix this on HEMTT side?

**Comment by PabstMirror** [comment]
I generally dislike any complex code in configs, `ARR_X` macros and mixing quotes types is a real pain to read and maintain.
We could just make it a new sqf function and have a simple `call` in config

**Review by PabstMirror — APPROVED** [review]

File: addons/interaction/functions/fnc_doRemoteControl.sqf [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#include "script_component.hpp"
+/*
+ * Author: ?
```
Comment (veteran29) [inline addons/interaction/functions/fnc_doRemoteControl.sqf:3]:
```suggestion
 * Author: Dystopian
```

ref: https://github.com/acemod/ACE3/pull/6632

**Review by veteran29 — CHANGES_REQUESTED** [review]

**Review by Dystopian — CHANGES_REQUESTED** [review]
also function description should be filled

File: addons/interaction/functions/fnc_doRemoteControl.sqf (hunk 2 above)
Comment (Dystopian) [inline addons/interaction/functions/fnc_doRemoteControl.sqf:3]:
```suggestion
 * Author: SilentSpike, joko
```
ref #1519

File: addons/interaction/ACE_ZeusActions.hpp [hunk 3]
```diff
@@ -35,19 +35,7 @@ class ACE_ZeusActions {
...
-                        side _x in [ARR_4(east,west,resistance,civilian)] \
-                        && !(isPlayer _x) \
-                    }, \
-                    objNull \
-                )]; \
-                bis_fnc_curatorObjectPlaced_mouseOver = [ARR_2('OBJECT',_unit)]; \
-                private _rc = group _target createUnit [ARR_5('ModuleRemoteControl_F',[ARR_3(0,0,0)],[],0,'NONE')]; \
-                _rc setVariable [ARR_2('BIS_fnc_initModules_disableAutoActivation',false)]; \
-            );
+            statement = QUOTE(call FUNC(doRemoteControl));
```
[the remark this answers is not in the document]
Comment (jokoho48) [reply addons/interaction/ACE_ZeusActions.hpp:38]: we use below also FUNC and in the hole project, we use FUNC in Configs and not DFUNC.
there are currently 738 uses of FUNCs in Configs and 119 DFUNCs and until now I never saw any issues reported with the use of FUNC.
also, I don't see any reason that macro should not work in configs.

**Review by PabstMirror — APPROVED** [review]

File: addons/interaction/ACE_ZeusActions.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (PabstMirror) [reply addons/interaction/ACE_ZeusActions.hpp:38]: I FUNC/DFUNC is only relevant for the stack tracing which I don't think gets much use at all in ace

**Review by PabstMirror — APPROVED** [review]

File: addons/interaction/ACE_ZeusActions.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (jonpas) [reply addons/interaction/ACE_ZeusActions.hpp:38]: Use DFUNC if you need it for callstack, we weren't very consistent with it from the start, so it doesn't matter at all now.

**Review by bux — APPROVED** [review]

**Pull request merged**